lib/src/rules/prefer_is_empty.dart

```diff
@@ -216,8 +216,8 @@ class _Visitor extends SimpleAstVisitor<void> {
 
     // Should be subtype of Iterable, Map or String.
     if (type == null ||
-        !DartTypeUtilities.implementsInterface(type, 'Iterable', 'dart.core') &&
-            !DartTypeUtilities.implementsInterface(type, 'Map', 'dart.core') &&
+        !type.implementsInterface('Iterable', 'dart.core') &&
+            !type.implementsInterface('Map', 'dart.core') &&
```

Comment:
Similarly here (and `isDartCoreMap`).

Reply:
I don't believe so. `MyMap` might implement `Map`, but the implementation of `isDartCoreMap` is

```dart
  return element2.name == "Map" && element2.library.isDartCore;
```

which would return false for elements which implement Map.

Reply:
You're right, I missed that.

Reply:
I'll leave these alone and fix the ones that aren't about subtyping.